MSRV

Currently, the `rust-version` of `exn` is 1.86 due to a trait upcasting coercion used.

<details><summary>Details</summary>
<p>

```rs
                              
 error[E0658]: cannot cast `dyn Error` to `dyn Any`, trait upcasting coercion is      
 experimental                                                                         
    --> /Users/elea/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/exn-0.2.1/src
 impls.rs:140:9                                                                       
     |                                                                                
 140 |         &*self.error                                                           
     |         ^^^^^^^^^^^^                                                           
     |                                                                                
     = note: see issue #65991 <https://github.com/rust-lang/rust/issues/65991> for mor
  information                                                                         
     = note: required when coercing `&(dyn Error + 'static)` into `&(dyn Any +        
 'static)`                                                                            
                                                                                      
 error[E0658]: cannot cast `dyn Error` to `dyn std::error::Error`, trait upcasting    
 coercion is experimental                                                             
    --> /Users/elea/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/exn-0.2.1/src
 impls.rs:145:9                                                                       
     |                                                                                
 145 |         &*self.error                                                           
     |         ^^^^^^^^^^^^                                                           
     |                                                                                
     = note: see issue #65991 <https://github.com/rust-lang/rust/issues/65991> for mor
  information                                                                         
     = note: required when coercing `&(dyn Error + 'static)` into `&dyn               
 std::error::Error`                                         
```

</p>
</details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -53,7 +53,7 @@ jobs:
     strategy:
       matrix:
         os: [ ubuntu-24.04, macos-14, windows-2022 ]
-        rust-version: [ "stable", "nightly" ]
+        rust-version: [ "1.85.0", "stable", "nightly" ]
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v6

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -22,6 +22,7 @@ homepage = "https://github.com/fast/exn"
 license = "Apache-2.0"
 readme = "README.md"
 repository = "https://github.com/fast/exn"
+rust-version = "1.85.0"
 
 [workspace.dependencies]
 # Workspace dependencies
@@ -34,8 +35,8 @@ insta = { version = "1.45.1" }
 which = { version = "8.0.0" }
 
 [workspace.lints.rust]
-unexpected_cfgs = { level = "deny", check-cfg = ['cfg(windows_test)'] }
 unknown_lints = "deny"
+unused_must_use = "deny"
 
 [workspace.lints.clippy]
 dbg_macro = "deny"

--- a/README.md
+++ b/README.md
@@ -2,6 +2,7 @@
 
 [![Crates.io][crates-badge]][crates-url]
 [![Documentation][docs-badge]][docs-url]
+[![MSRV 1.85][msrv-badge]](https://www.whatrustisit.com)
 [![Apache 2.0 licensed][license-badge]][license-url]
 [![Build Status][actions-badge]][actions-url]
 
@@ -9,6 +10,7 @@
 [crates-url]: https://crates.io/crates/exn
 [docs-badge]: https://docs.rs/exn/badge.svg
 [docs-url]: https://docs.rs/exn
+[msrv-badge]: https://img.shields.io/badge/MSRV-1.85-green?logo=rust
 [license-badge]: https://img.shields.io/crates/l/exn
 [license-url]: LICENSE
 [actions-badge]: https://github.com/fast/exn/workflows/CI/badge.svg
@@ -29,6 +31,12 @@ Read the online documents at https://docs.rs/exn.
 This crate is `no_std` compatible, while the `alloc` crate is still required for heap allocations.
 
 It is worth noting that `no_std` support is a nice-to-have feature, and can be dropped if it blocks other important features in the future. Before 1.0, once `exn` APIs settle down, the decision on whether to keep `no_std` as a promise will be finalized.
+
+## Minimum Rust version policy
+
+This crate is built against the latest stable release, and its minimum supported rustc version is 1.85.0.
+
+The policy is that the minimum Rust version required to use this crate can be increased in minor version updates. For example, if version 1.0 requires Rust 1.60.0, then version 1.0.z for all values of z will also require Rust 1.60.0 or newer. However, version 1.y for y > 0 may require a newer minimum version of Rust.
 
 ## License
 

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -13,5 +13,5 @@
 # limitations under the License.
 
 [toolchain]
-channel = "stable"
+channel = "1.85.0"
 components = ["cargo", "rustfmt", "clippy", "rust-analyzer"]

